api/src/controllers/extensions.ts

```diff
@@ -64,8 +66,8 @@ router.get(
 			query.offset = Number(offset);
 		}
 
-		if (typeof by === 'string') {
-			query.by = by;
+		if (filter?.by) {
+			query.by = filter.by._eq as string;
 		}
```

Comment:
Could look something like this:
- get `type` via filter too, as already mentioned
- proper checks, so no assertions required 
```suggestion
		if (filter) {
			if ('by' in filter && '_eq' in filter['by'] && typeof filter['by']._eq === 'string') {
				query.by = filter['by']._eq;
			}

			if ('type' in filter && '_eq' in filter['type'] && typeof filter['type']._eq === 'string') {
				if (isIn(filter['type']._eq, EXTENSION_TYPES) === false) {
					throw new ForbiddenError();
				}
	
				query.type = filter['type']._eq;
			}
		}
```

Could of course be refined a little (e.g. a small function for obtaining the two filter values?)